skins/oasis/modules/UserPagesHeaderController.class.php

```diff
@@ -71,325 +74,166 @@ public static function getUserName( Title $title, $namespaces, $fallbackToGlobal
 
 	/**
 	 * Get list of links for given username to be shown as tabs
+	 * @param string $userName
+	 * @return array Array of tabs - each tab is an array containing an HTML link, the data-id attribute and whether it is selected
 	 */
 	private function getTabs( $userName ) {
 		wfProfileIn( __METHOD__ );
-		global $wgTitle, $wgEnableWikiaFollowedPages;
 
-		$tabs = array();
-		$namespace = $wgTitle->getNamespace();
+		$tabs = [];
+		$namespace = $this->wg->Title->getNamespace();
 
 		// profile
-		$tabs[] = array(
-				'link' => Wikia::link( Title::newFromText( $userName, NS_USER ), wfMsg( 'profile' ) ),
-				'selected' => ($namespace == NS_USER),
-				'data-id' => 'profile',
-				);
+		$tabs[] = [
+			'link' => Linker::link( Title::newFromText( $userName, NS_USER ), wfMessage( 'profile' )->escaped() ),
+			'selected' => ( $namespace == NS_USER ),
+			'data-id' => 'profile',
+		];
 
 		// talk
-		$tabs[] = array(
-				'link' => Wikia::link( Title::newFromText( $userName, NS_USER_TALK ), wfMsg( 'talkpage' ) ),
-				'selected' => ( $namespace == NS_USER_TALK ),
-				'data-id' => 'talk',
-				);
+		$tabs[] = [
+			'link' => Linker::link( Title::newFromText( $userName, NS_USER_TALK ), wfMessage( 'talkpage' )->escaped() ),
+			'selected' => ( $namespace == NS_USER_TALK ),
+			'data-id' => 'talk',
+		];
 
 		// blog
-		if( defined( 'NS_BLOG_ARTICLE' ) && !User::isIP( $this->userName ) ) {
-			$tabs[] = array(
-				'link' => Wikia::link( Title::newFromText( $userName, NS_BLOG_ARTICLE ), wfMsg( 'blog-page' ), array(), array(), 'known' ),
+		if ( defined( 'NS_BLOG_ARTICLE' ) && !User::isIP( $this->userName ) ) {
+			$tabs[] = [
+				'link' => Linker::linkKnown( Title::newFromText( $userName, NS_BLOG_ARTICLE ), wfMessage( 'blog-page' )->escaped() ),
 				'selected' => ( $namespace == NS_BLOG_ARTICLE ),
 				'data-id' => 'blog',
-			);
+			];
 		}
 
 		// contribs
-		$tabs[] = array(
-			'link' => Wikia::link( SpecialPage::getTitleFor( "Contributions/{$userName}" ), wfMsg( 'contris_s' ) ),
-			'selected' => ( $wgTitle->isSpecial( 'Contributions' ) ),
+		$tabs[] = [
+			'link' => Linker::linkKnown( SpecialPage::getTitleFor( "Contributions/{$userName}" ), wfMessage( 'contris_s' )->escaped() ),
+			'selected' => ( $this->wg->Title->isSpecial( 'Contributions' ) ),
 			'data-id' => 'contribs',
-		);
+		];
 
 		if ( self::isItMe( $userName ) ) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Use "static" keyword instead of "self". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS2037)